### PR TITLE
Add type for new "ref" proptype in airbnb-prop-type

### PR DESCRIPTION
**Pull request by ahuth**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/airbnb/prop-types/pull/54
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/airbnb-prop-types/index.d.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
-// Type definitions for airbnb-prop-types 2.11
+// Type definitions for airbnb-prop-types 2.13
 // Project: https://github.com/airbnb/prop-types
 // Definitions by: Miles Johnson <https://github.com/milesj>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
 import * as PropTypes from 'prop-types';
+import * as React from 'react';
```
Comment (milesj) [inline types/airbnb-prop-types/index.d.ts]: We purposefully avoided coupling the `@types/react` package (as seems by all the interfaces below). Can you add a `RefLike` instead?

Comment (ahuth) [reply types/airbnb-prop-types/index.d.ts]: Sure

File: types/airbnb-prop-types/index.d.ts [hunk 2]
```diff
@@ -24,6 +24,12 @@ export type ReactFunctionComponentLike = (...args: any[]) => PropTypes.ReactNode
 
 export type ReactTypeLike = string | ReactClassComponentLike | ReactFunctionComponentLike;
 
+export interface ReactRefLike<T> {
+    readonly current: T | null;
+}
+
+export type ReactLegacyRefLike<T> = (instance: T | null) => void | ReactRefLike<T> | null;
```
Comment (ahuth) [inline types/airbnb-prop-types/index.d.ts]: Not totally sure we want `| null` or not. Included it because React's definition for ref does.

**Comment by ahuth** [comment]
Thanks for looking at this @milesj. Can you take a look at this again, please?

File: types/airbnb-prop-types/index.d.ts (hunk 2 above)
Comment (ahuth) [reply types/airbnb-prop-types/index.d.ts]: Meh, decided to remove it

File: types/airbnb-prop-types/index.d.ts [hunk 3]
```diff
@@ -24,6 +24,12 @@ export type ReactFunctionComponentLike = (...args: any[]) => PropTypes.ReactNode
 
 export type ReactTypeLike = string | ReactClassComponentLike | ReactFunctionComponentLike;
 
+export interface ReactRefLike<T> {
+    readonly current: T | null;
+}
+
+export type ReactLegacyRefLike<T> = (instance: T | null) => void | ReactRefLike<T>;
```
Comment (milesj) [inline types/airbnb-prop-types/index.d.ts]: I think you want `((instance: T | null) => void) | ReactRefLike<T>`, which is a union of either. Currently it's a return type of either.

Comment (ahuth) [reply types/airbnb-prop-types/index.d.ts]: Yeah, you're right. Nice catch.

I avoided this while testing because the order was flipped

**Comment by ahuth** [comment]
Alright @milesj, I think this is sort of, kind of, probably, maybe right now

**Review by milesj — APPROVED** [review]
Looks good, thanks!

**Pull request merged**